src/components/ChatEntry.js

```diff
@@ -1,22 +1,52 @@
-import React from 'react';
+//import React, { useState } from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+import {useState} from 'react';
+
 
 const ChatEntry = (props) => {
+  const[likeButton,setLikeButton]=useState('🤍');
+    const clickedToggle=()=>{
+
+    if(likeButton==='🤍'){
+      setLikeButton('❤️')
+      props.updateLikes(true)
+    }else{
+      setLikeButton('🤍')
+      props.updateLikes(false)
+    }
+  }
+  
+  
   return (
     <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
-      <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+      <h2 className="entry-name">{props.sender}</h2>
+      <section className="entry-bubble">{props.body}
+        <p></p>
+        <p className="entry-time">
+          <TimeStamp time={props.timeStamp}/>
+        </p>
+        <button className="like" onClick={clickedToggle}>{likeButton}</button>
```

Comment:
emember that we have passed down a function that is supposed to handle this for us. If we are having to rewrite this function, we aren't effectively lifting state up.